test/helm_task_test.go

```diff
@@ -0,0 +1,310 @@
+	if err != nil {
+		t.Errorf("Error getting service at %s %s", helmDeployServiceName, err)
+	}
+	var serviceIp string
+	ingress := k8sService.Status.LoadBalancer.Ingress
+	if len(ingress) > 0 {
+		serviceIp = ingress[0].IP
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://%s:8080", serviceIp))
+	if err != nil {
+		t.Errorf("Error reaching service at http://%s:8080 %s", serviceIp, err)
+	}
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		t.Errorf("Error from service at http://%s:8080 %s", serviceIp, err)
+	}
+}
+
+func getGoHelloworldGitResource(namespace string) *v1alpha1.PipelineResource {
+	return &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourceResourceName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: v1alpha1.PipelineResourceTypeGit,
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/pivotal-nader-ziada/gohelloworld",
+				},
+			},
+		},
+	}
+}
+
+func getCreateImageTask(namespace string, t *testing.T) *v1alpha1.Task {
+	// according to knative/test-infra readme (https://github.com/knative/test-infra/blob/13055d769cc5e1756e605fcb3bcc1c25376699f1/scripts/README.md)
+	// the KO_DOCKER_REPO will be set with according to the porject where the cluster is created
+	// it is used here to dunamically get the docker registery to push the image to
+	dockerRepo := os.Getenv("KO_DOCKER_REPO")
+	if dockerRepo == "" {
+		t.Fatalf("KO_DOCKER_REPO env variable is required")
+	}
+
+	imageName = fmt.Sprintf("%s/%s", dockerRepo, sourceImageName)
+
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      createImageTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Resources: []v1alpha1.TaskResource{
+					v1alpha1.TaskResource{
+						Name: sourceResourceName,
+						Type: v1alpha1.PipelineResourceTypeGit,
+					},
+				},
+			},
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Steps: []corev1.Container{{
+					Name:  "kaniko",
+					Image: "gcr.io/kaniko-project/executor",
+					Args: []string{"--dockerfile=/workspace/Dockerfile",
+						fmt.Sprintf("--destination=%s", imageName),
+					},
+				}},
+			},
+		},
+	}
+}
+
+func getHelmDeployTask(namespace string) *v1alpha1.Task {
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      helmDeployTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Resources: []v1alpha1.TaskResource{
+					v1alpha1.TaskResource{
+						Name: sourceResourceName,
+						Type: v1alpha1.PipelineResourceTypeGit,
+					},
+				},
+				Params: []v1alpha1.TaskParam{{
+					Name: "pathToHelmCharts",
+				}, {
+					Name: "image",
+				}, {
+					Name: "chartname",
+				}},
+			},
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Steps: []corev1.Container{{
+					Name:  "helm-init",
+					Image: "alpine/helm",
+					Args:  []string{"init"},
+				},
```

Comment:
i wonder if the teardown step in the test should invoke a Task or something to remove `helm`, so that ppl who run this test aren't left with helm running in their clusters (or will the namespace take care of that?)

Reply:
doesn't the cluster get deleted after the test run in Prow?

Reply:
> I was worried if if things fail running another task might not work for whatever reason and I wasn't sure how else to remove it

That's true, so a couple options:
- Try to do it anyway but don't actually fail anything if the teardown fails (just log it)
- Try to do teardown without using Pipeline/Task, e.g. call out to kubectl

> doesn't the cluster get deleted after the test run in Prow?

Yeah I think so! So we're fine for Prow, I'm more concerned about ppl running this on their own clusters.

Reply:
will get the test green first then add the cleanup